sonarr/indexer.go

```diff
@@ -0,0 +1,234 @@
+package sonarr
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"path"
+	"strconv"
+
+	"golift.io/starr"
+)
+
+type IndexerInput struct {
+	EnableAutomaticSearch   bool           `json:"enableAutomaticSearch"`
+	EnableInteractiveSearch bool           `json:"enableInteractiveSearch"`
+	EnableRss               bool           `json:"enableRss"`
+	Priority                int64          `json:"priority"`
+	ID                      int64          `json:"id,omitempty"`
+	ConfigContract          string         `json:"configContract"`
+	Implementation          string         `json:"implementation"`
+	Name                    string         `json:"name"`
+	Protocol                string         `json:"protocol"`
+	Fields                  IndexerFields  `json:"fields"`
+	Tags                    []*starr.Value `json:"tags"`
+}
+
+type IndexerOutput struct {
```

Comment:
That was my initial approach, I thought you preferred to keep them separated.

Reply:
I thought your initial approach was 1 struct? Maybe I'm misremembering.

Reply:
yes, my initial approach is one struct only. If it's fine for you, I'll go back to that approach.

Reply:
I'm suggesting embedding the input struct into the output struct. It's an idea right now, and not something I think you need to do, but something we should consider.

Like this:
```
type IndexerOutput struct {
 	IndexerInput
 	ImplementationName      string         `json:"implementationName"`
 	InfoLink                string         `json:"infoLink"`
 	SupportsRss             bool           `json:"supportsRss"`
 	SupportsSearch          bool           `json:"supportsSearch"`
}
```

Reply:
Ok, got it. It really depends on the `Field` struct, if we want to have a `FieldInput` and a `FieldOutput` struct, we'll need to keep `IndexerInput` and `IndexerOutput` separated, otherwise we can do as you suggested.

Reply:
I suppose I'm going to recommend splitting it all up then. The inputs and outputs are just different enough that it makes sense to decouple them.